lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -943,6 +766,95 @@ struct Deduper {
   DenseMap<Operation *, ModuleNamespace> moduleNamespaces;
 };
 
+//===----------------------------------------------------------------------===//
+// Fixup
+//===----------------------------------------------------------------------===//
+
+/// This fixes up connects when the field names of a bundle type changes.  It
+/// finds all fields which were previously bulk connected and legalizes it
+/// into a connect for each field.
+template <typename T>
+void fixupConnect(ImplicitLocOpBuilder &builder, Value dst, Value src) {
+  // If the types already match we can emit a connect.
+  auto dstType = dst.getType();
+  auto srcType = src.getType();
+  if (dstType == srcType) {
+    builder.create<T>(dst, src);
+    return;
+  }
+  // It must be a bundle type and the field name has changed. We have to
+  // manually decompose the bulk connect into a connect for each field.
+  auto dstBundle = dstType.cast<BundleType>();
+  auto srcBundle = srcType.cast<BundleType>();
+  for (unsigned i = 0; i < dstBundle.getNumElements(); ++i) {
+    auto dstField = builder.create<SubfieldOp>(dst, i);
+    auto srcField = builder.create<SubfieldOp>(src, i);
+    if (dstBundle.getElement(i).isFlip) {
+      std::swap(srcBundle, dstBundle);
+      std::swap(srcField, dstField);
+    }
+    fixupConnect<T>(builder, dstField, srcField);
+  }
+}
+
+/// Replaces a ConnectOp or StrictConnectOp with new bundle types.
+template <typename T>
+void fixupConnect(T connect) {
+  ImplicitLocOpBuilder builder(connect.getLoc(), connect);
+  fixupConnect<T>(builder, connect.dest(), connect.src());
+  connect->erase();
+}
+
+/// When we replace a bundle type with a similar bundle with different field
+/// names, we have to rewrite all the code to use the new field names. This
+/// mostly affects subfield result types and any bulk connects.
+// NOLINTNEXTLINE(misc-no-recursion)
+void fixupReferences(Value oldValue, Type newType) {
```

Comment:
This function had a mildy complicated change and wasn't just c/p'd. This parameters changed from `newValue` and `oldType` to `oldValue` and `newType`. Previously, it would assume the current value had already had its type updated and that we just needed to update the uses.  Now it must update the type of the current operation, to cover instance operations, and then adds uses to the worklist.